lib/shared-functions.sh

```diff
@@ -180,11 +180,11 @@ function check_sharelatex_env_vars() {
 function read_variable() {
   local name=$1
   grep -E "^$name=" "$TOOLKIT_ROOT/config/variables.env" \
-  | sed -r "s/^$name=([\"']?)(.+)\1\$/\2/"
+  | sed -r "s/^$name=([\"']?)([^\"']+)[\"']?$/\2/"
```

Comment:
This will not trim in case there are quotes inside the value, e.g. `foo='Instance managed by "Department"'`.

before:
```
$ name=foo; cat | sed -r "s/^$name=([\"']?)(.+)\1\$/\2/"
foo='Instance managed by "Department"'

// output
Instance managed by "Department"
```

after:
```
$ name=foo; cat | sed -r "s/^$name=([\"']?)([^\"']+)[\"']?$/\2/"
foo='Instance managed by "Department"'

// output
foo='Instance managed by "Department"'
```

Using a non-greedy matching group instead of the "non-quote"-group in the middle does not work either:
```
$ name=foo; cat | sed -r "s/^$name=([\"']?)(.+?)[\"']?$/\2/"
foo='Instance managed by "Department"'

// output
Instance managed by "Department"'
                                ^-- unmatched quote
```

Reply:
How about using bash to do the parsing in a subshell?

```bash
function read_variable() {
  local name=$1
  (
    source "$TOOLKIT_ROOT/config/variables.env"
    echo "${!name:-}"
  )
}
```

Reply:
@das7pad just wanted to check-in, if we can merge this PR or is there anything else I need to do prior to that? Thanks.